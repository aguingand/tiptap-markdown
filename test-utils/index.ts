
export { parse } from './parse.js';
export { serialize } from './serialize';

export function dedent(str: TemplateStringsArray) {
    return str[0]!.replace(/^\s*/gm, '');
}
