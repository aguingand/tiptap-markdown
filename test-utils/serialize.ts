import { Node as ProseMirrorNode } from "@tiptap/pm/model";
import { Content, getHTMLFromFragment, createNodeFromContent } from "@tiptap/core";
import { createEditor, TestEditorOptions } from "./editor";

export function serialize(content: Content, options: TestEditorOptions = {}) {
    const editor = createEditor(options);
    const doc = createNodeFromContent(content, editor.schema, {
        parseOptions: {
            preserveWhitespace: true,
        }
    }) as ProseMirrorNode;
    const html = getHTMLFromFragment(doc.content, editor.schema);

    return editor.storage.markdown.serializer.serialize(html);
}
