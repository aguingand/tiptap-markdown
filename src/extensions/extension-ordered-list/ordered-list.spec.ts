import { expect, test } from "vitest";
import { parse, serialize } from "../../../test-utils";


test('parse markdown', () => {
    expect(parse('1. example1\n2. example2')).toMatchSnapshot();
});

test('parse html', () => {
    expect(parse('<ol><li>example1</li><li>example2</li></ol>')).toMatchSnapshot();
});

test('serialize', () => {
    expect(serialize('<ol><li>example1</li><li>example2</li></ol>'))
        .toEqual('1. example1\n2. example2\n');
    expect(serialize('<ol start="10"><li>example1</li><li>example2</li></ol>'))
        .toEqual('10. example1\n11. example2\n');
});

test('serialize adjacent', () => {
    expect(serialize('<ol><li>example1</li></ol><ol><li>example2</li></ol><ol><li>example3</li></ol>'))
        .toEqual('1. example1\n\n1) example2\n\n1. example3\n');
})
