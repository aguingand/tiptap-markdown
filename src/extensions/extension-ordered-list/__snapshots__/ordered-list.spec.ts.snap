// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse html 1`] = `
[
  {
    "attrs": {
      "start": 1,
      "tight": true,
      "type": undefined,
    },
    "content": [
      {
        "content": [
          {
            "content": [
              {
                "text": "example1",
                "type": "text",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "listItem",
      },
      {
        "content": [
          {
            "content": [
              {
                "text": "example2",
                "type": "text",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "listItem",
      },
    ],
    "type": "orderedList",
  },
]
`;

exports[`parse markdown 1`] = `
[
  {
    "attrs": {
      "start": 1,
      "tight": true,
      "type": undefined,
    },
    "content": [
      {
        "content": [
          {
            "content": [
              {
                "text": "example1",
                "type": "text",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "listItem",
      },
      {
        "content": [
          {
            "content": [
              {
                "text": "example2",
                "type": "text",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "listItem",
      },
    ],
    "type": "orderedList",
  },
]
`;
